### PR TITLE
Document a gotcha with `additionalProperties`

### PR DESCRIPTION
**Pull request by loganvolkers**
Relates to #848 #902 #992 (and maybe others).

### Reasons for making this change

It seems like a lot of people are running into the same problem, this will help them avoid it earlier and be more clear about the current limitations of the library.

### Checklist

* [x] **I'm updating documentation**
  - [x] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [ ] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by glasserc** [comment]
Thanks for submitting this, and sorry for taking so long to respond to it. This reads like a pretty mild warning but it almost sounds like it's describing a bug. Is it actually a bug? Is using `additionalProperties: false` incorrect somehow? If not, is there a workaround -- something else that should be used?

**Comment by loganvolkers** [comment]
#848 has a pretty good example of the behaviour. Using `additionalProperties: false` will produce incorrect results and break things. There may be a fix for this that would delete additional properties from the output, but would be hard to work.

Either way, it is a limitation that seems to be commonly encountered, so it should be more clearly documented.

File: README.md [hunk 1]
```diff
@@ -1744,6 +1744,7 @@ This component follows [JSON Schema](http://json-schema.org/documentation.html)
 * `anyOf`, `allOf`, and `oneOf`, or multiple `types` (i.e. `"type": ["string", "array"]`
     Nobody yet has come up with a PR that adds this feature with a simple and easy-to-understand UX.
     You can use `oneOf` with [schema dependencies](#schema-dependencies) to dynamically add schema properties based on input data but this feature does not bring general support for `oneOf` elsewhere in a schema.
+* `"additionalProperties":false` when used with [schema dependencies](#schema-dependencies) will not remove the extra properties and produce invalid schemas. It is recommended to avoid setting `"additionalProperties":false`. See [#848](https://github.com/mozilla-services/react-jsonschema-form/issues/848) [#902](https://github.com/mozilla-services/react-jsonschema-form/issues/902) [#992](https://github.com/mozilla-services/react-jsonschema-form/issues/992)
```
Comment (epicfaace) [inline README.md]: Can you change the wording here? I can't tell if this means "will not remove the extra properties and will produce invalid schemas" or "will not remove the extra properties and will not produce invalid schemas"

Comment (epicfaace) [reply README.md]: Also, can you make your change in the `docs` directory instead of the README? Thanks

Comment (epicfaace) [reply README.md]: Ok, that's great! Let me know what kind of features / bugs you come across; I'll try to be as responsive as possible to new ideas / PRs.

**Pull request closed without merging**